### PR TITLE
feat(self-managed): support custom tenant providers in zeebe

### LINKED ISSUE #3029: Document support for a custom tenant-providing interceptor in Zeebe SM
### Description
Document support for a custom tenant-providing interceptor in Zeebe:

```[tasklist]
### Tasks
- [x] Extend the [Interceptor docs](https://docs.camunda.io/docs/self-managed/zeebe-deployment/zeebe-gateway/interceptors/) with a section on *Implementing a tenant-providing interceptor*
- [x] Update the MultiTenancy docs for Zeebe to state that multi-tenancy in Zeebe is independent from Identity, if Zeebe is used as a standalone product
- [x] If Zeebe is used as a standalone product, and multi-tenancy is enabled, users can implement a custom tenant-providing interceptor. A link to the Interceptor docs should be provided.
- [x] If the full Camunda 8 stack is used, then a custom tenant-providing interceptor can't be used, and Identity must be configured since other Camunda 8 components must use Identity.
```

### Additional context
Related to https://github.com/camunda/zeebe/issues/14285

### PR DESCRIPTION
**Pull request by koevskinikola**
## Description

Document how users can intergrate their own tenant-management component with Zeebe. Furthermore, document how the support for custom tenant provider fits with the multi-tenancy concept in the Camunda 8 stack.

closes #3029 

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/self-managed/concepts/multi-tenancy.md [hunk 1]
```diff
@@ -29,6 +29,9 @@ Unexpected behavior may occur if multi-tenancy is only enabled in some component
 :::caution
 
 Multi-tenancy is currently only available for Camunda 8 Self-Managed with authentication enabled [through Identity](../../../self-managed/identity/what-is-identity/).
+For a standalone Zeebe instance, multi-tenancy is currently available with authentication enabled [through Identity](../../../self-managed/identity/what-is-identity/)
+or by [integrating an external tenant-managing component](../zeebe-deployment/zeebe-gateway/interceptors.md#implementing-a-tenant-providing-interceptor)
+with Zeebe.
```
Comment (koevskinikola) [inline docs/self-managed/concepts/multi-tenancy.md:34]: :question: @christinaausley I'm interested about your opinon on mentioning this feature in the conceptual multi-tenancy docs, as this document is mainly about Camunda 8, and not really about a standalone Zeebe instance. 

However, a standalone Zeebe instance can still be considered a self-managed setup.

Comment (christinaausley) [reply docs/self-managed/concepts/multi-tenancy.md:34]: I'm fine with this 👍

File: docs/self-managed/zeebe-deployment/zeebe-gateway/interceptors.md [hunk 2]
```diff
@@ -75,6 +75,64 @@ all incoming calls to the target broker, but it would also be possible to stop
 the message from interception by other interceptors and even to block it from
 dispatch to the broker.
 
+### Implementing a tenant-providing interceptor
+
+:::note
+Camunda 8 relies on Identity for tenant management. Tenant-providing interceptors are only compatible with Zeebe, and
```
Comment (christinaausley) [inline docs/self-managed/zeebe-deployment/zeebe-gateway/interceptors.md:81]: I would link to https://docs.camunda.io/docs/next/self-managed/identity/user-guide/tenants/managing-tenants/ here.

File: docs/self-managed/zeebe-deployment/zeebe-gateway/interceptors.md [hunk 3]
```diff
@@ -75,6 +75,64 @@ all incoming calls to the target broker, but it would also be possible to stop
...
+should only be used when Zeebe is used as a standalone application.
```
Comment (christinaausley) [inline docs/self-managed/zeebe-deployment/zeebe-gateway/interceptors.md:82]: If Zeebe is used as a standalone application and a user doesn't have access to Helm, do they still need to ensure they have enabled multi-tenancy across all components?

**Review by christinaausley — COMMENTED** [review]
Two comments for clarity 👍

File: docs/self-managed/concepts/multi-tenancy.md [hunk 4]
```diff
@@ -94,3 +97,11 @@ Camunda 8 data produced from these resources will belong to the same tenant. The
 example on how tenant ownership is inherited.
 
 ![Tenant ownership inheritance diagram](img/multi-tenancy.png)
+
+## Unsupported features
+
+Multi-tenancy only works for Self-Managed installations with authentication enabled [through Identity](../../../self-managed/identity/what-is-identity/).
+
+For a standalone Zeebe instance, multi-tenancy is currently available with authentication enabled [through Identity](../../../self-managed/identity/what-is-identity/)
+or by [integrating an external tenant-managing component](../zeebe-deployment/zeebe-gateway/interceptors.md#implementing-a-tenant-providing-interceptor)
```
Comment (nicpuppa) [inline docs/self-managed/concepts/multi-tenancy.md:106]: ❓ Why the title is `Unsupported features` ?

**Review by nicpuppa — DISMISSED** [review]
I have just a small question but everything looks good to me 👍

File: docs/self-managed/zeebe-deployment/zeebe-gateway/interceptors.md (hunk 3 above)
Comment (koevskinikola) [reply docs/self-managed/zeebe-deployment/zeebe-gateway/interceptors.md:82]: No, they won't have to enable multi-tenancy across all components. When Zeebe is used as a standalone application, the assumption is that no other Camunda 8 components are used.

If a user decides to use Zeebe with a custom tenant-providing interceptor and with other C8 components, then multi-tenancy will not work properly, as the other components will rely on Identity for tenant management. As a result, users won't be able to access their data correctly in Operate, Optimize, Tasklist, etc.

File: docs/self-managed/concepts/multi-tenancy.md (hunk 4 above)
Comment (koevskinikola) [reply docs/self-managed/concepts/multi-tenancy.md:106]: Good point. What I added is not really an unsupported feature. Maybe a separate section called "Standalone Zeebe usage" would work better.

I'm open for other suggestions for the heading :smile:

Comment (nicpuppa) [reply docs/self-managed/concepts/multi-tenancy.md:106]: Yeah, but what is the point to repeat what wrote at lines 32-34 ? 😄 
If we want to add a specific section for standalone zeebe, I will remove the lines 32-34 and create a separate section `Standalone Zeebe usage`, but without the Self-Managed explanation

Comment (koevskinikola) [reply docs/self-managed/concepts/multi-tenancy.md:106]: Although it's not supported for the full Camunda 8 stack. I'll see how I can rephrase a mention in this section.

File: docs/self-managed/zeebe-deployment/zeebe-gateway/interceptors.md (hunk 3 above)
Comment (koevskinikola) [reply docs/self-managed/zeebe-deployment/zeebe-gateway/interceptors.md]: @christinaausley I documented the above with https://github.com/camunda/camunda-docs/pull/3100/commits/9389f0b14ea83291c24b878556b723bbb0c2f7fa

**Comment by koevskinikola** [comment]
@nicpuppa @christinaausley have a look at my latest changes.

File: docs/self-managed/concepts/multi-tenancy.md [hunk 5]
```diff
@@ -94,3 +97,19 @@ Camunda 8 data produced from these resources will belong to the same tenant. The
...
+
+- Authentication enabled [through Identity](../../../self-managed/identity/what-is-identity/)
+- By [integrating an external tenant-managing component](../zeebe-deployment/zeebe-gateway/interceptors.md#implementing-a-tenant-providing-interceptor)
+  with Zeebe.
+
+Note that it's not possible to use multi-tenancy on the full Camunda 8 stack when
+[integrating an external tenant-managing component](../zeebe-deployment/zeebe-gateway/interceptors.md#implementing-a-tenant-providing-interceptor)
+in Zeebe, as the remaining Camunda 8 components don't support this setup.
+
+## Unsupported features
```
Comment (nicpuppa) [inline docs/self-managed/concepts/multi-tenancy.md:113]: This section is still not clear to me 😄 Which are the unsupported features ?

Comment (abbasadel) [reply docs/self-managed/concepts/multi-tenancy.md:113]: It was clear to me that using the external tenant management will not work with Camunda 8 components.

**Review by abbasadel — APPROVED** [review]
LGTM 👍🏽

File: docs/self-managed/concepts/multi-tenancy.md (hunk 5 above)
Comment (koevskinikola) [reply docs/self-managed/concepts/multi-tenancy.md:113]: @nicpuppa that section is addressing the unsupported behavior of Camunda 8 as a complete product.

So we mention that multi-tenancy will only work if Identity is used and enabled.

I'm also expanding that section with [this PR](https://github.com/camunda/camunda-docs/pull/3101).

Comment (nicpuppa) [reply docs/self-managed/concepts/multi-tenancy.md:113]: It is ok for me, thanks for clarifying 👍

**Review by christinaausley — APPROVED** [review]

**Comment by koevskinikola** [comment]
@christinaausley I'm not sure what we need to crosslink from https://github.com/camunda/camunda-docs/pull/3048#issuecomment-1866361553. That PR deals more with the Operate/Tasklist/Optimize docs.

It's currently not possible to use a custom tenant provider component in C8 apps other than Zeebe.

Can you expand on what you meant?

**Comment by christinaausley** [comment]
@koevskinikola I added this as a note to myself to double check -- this looks good!

**Comment by koevskinikola** [comment]
Thanks for the quick reply @christinaausley. In that case, I'll merge the PR.

**Pull request merged**